### ISSUE TITLE
Storing Training Log Loss and Training Accuracy Results 

### ISSUE THREAD

**Issue by Shadz13** [massquantity/LibRecommender#44]
Currently trying find the following the library scripts:
1) Training log loss or loss per epoch
2) Training accuracy per epoch

I noticed during the runs the training log loss is printed on the screen but cant seem to find in the evaluation.py script.
Below is a snapshot of the training log loss used on an example dummy sample set :
![training_log](https://user-images.githubusercontent.com/42578613/109207094-66a5f100-77b1-11eb-8d78-836d6fc124f3.PNG)

There is a training log loss value, but since it did not print the word VALUE I assumed its not the result printed in the window.

```
def print_metrics_ranking(self,df,metrics, y_prob=None, y_true=None, y_reco_list=None,
                          y_true_list=None, users=None, k=3, train=True):
    eval_df=pd.DataFrame()
    self.epoch_num = self.epoch_num +1
    if train:
        for m in metrics:
            if m in ["log_loss", "loss"]:
                log_loss_ = log_loss(y_true, y_prob, eps=1e-7)
                print(f"\t train log_loss: {log_loss_:.4f}")
                eval_df['train_log_loss']=[log_loss_]
                print('VALUE')
```

**Comment by massquantity** [comment]
Training loss appears in `libreco/algorithms/base.py`, line 333-337. 
Since during training, the process needs to calculate the training loss to update the model. So if we calculate the training loss again in `evaluate.py`,  the training loss will be computed twice, which  is inefficient.

**Comment by Shadz13** [comment]
Thanks, I may have overlooked it. Saved!

Can this training loss be compared directly to the eval loss recorded in the evaluation script or does it need be converted?

**Comment by massquantity** [comment]
The training_loss is computed using [tf.nn.sigmoid_cross_entropy_with_logits](https://www.tensorflow.org/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits), and the eval loss is computed using [sklearn.metrics.log_loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html).
The math equations are the same, so if you trust the implementations of both TensorFlow ans Sklearn, then they can be compared.
